primitives/polkadot-core/src/lib.rs

```diff
@@ -51,29 +51,36 @@ pub mod parachains;
 ///
 /// Right now both Kusama and Polkadot aim to have around 1000 validators. Let's be safe here and
 /// take twice as much here.
-pub const MAX_AUTHORITIES_COUNT: u32 = 2_048;
+pub const MAX_AUTHORITIES_COUNT: u32 = 1_256;
 
 /// Reasonable number of headers in the `votes_ancestries` on Polkadot-like chains.
 ///
 /// See [`bp_header_chain::ChainWithGrandpa`] for more details.
-pub const REASONABLE_HEADERS_IN_JUSTIFICATON_ANCESTRY: u32 = 8;
+///
+/// This value comes from recent (February, 2023) Kusama and Polkadot headers. There are no
+/// justifications with any additional headers in votes ancestry, so reasonable headers may
```

Comment:
I would be interested in learning more about this. Do you have a link to the code where this value comes from ?

Reply:
Sure. Here's the code (if you need the same for Polkadot - just change the host to `rpc.polkadot.io`):
```rust
async_std::task::block_on(async {
		use codec::{Decode, Encode};
		use relay_substrate_client::BlockWithJustification;
		use sp_finality_grandpa::{ConsensusLog, GrandpaJustification, GRANDPA_ENGINE_ID};
		use sp_runtime::traits::Header;

		let kusama = relay_substrate_client::Client::<relay_kusama_client::Kusama>::new(
			relay_substrate_client::ConnectionParams {
				host: "kusama-rpc.polkadot.io".into(),
				port: 443,
				secure: true,
				..Default::default()
			},
		)
		.await;

		let mut total_maximal_headers = 0;
		let mut total_maximal_header_size = 0;
		let mut maximal_header_size = 0;

		let mut total_average_headers = 0;
		let mut total_average_header_size = 0;
		let mut maximal_average_header_size = 0;

		let mut total_justifications = 0;
		let mut total_justifications_size = 0;
		let mut max_justification_ancestry = 0;

		let mut best_finalized = kusama.best_finalized_header().await.unwrap();
		while *best_finalized.number() > 0 {
			let block = kusama.get_block(Some(best_finalized.hash())).await.unwrap();
			let header = block.header();
			best_finalized = kusama.header_by_hash(*header.parent_hash()).await.unwrap();

			let is_maximal_header = header
				.digest()
				.convert_first(|log| log.consensus_try_to(&GRANDPA_ENGINE_ID))
				.and_then(|log: ConsensusLog<bp_polkadot_core::BlockNumber>| match log {
					ConsensusLog::ScheduledChange(change) => Some(change),
					_ => None,
				})
				.is_some();

			let header_size = header.encode().len();
			if is_maximal_header {
				total_maximal_headers += 1;
				total_maximal_header_size += header.encode().len();
				if header_size > maximal_header_size {
					maximal_header_size = header_size;
				}
			} else {
				total_average_headers += 1;
				total_average_header_size += header_size;
				if header_size > maximal_average_header_size {
					println!("{:?}", header);

					maximal_average_header_size = header_size;
				}
			}

			let justification = block.justification(GRANDPA_ENGINE_ID);
			if let Some(justification) = justification {
				let justification = GrandpaJustification::<bp_polkadot_core::Header>::decode(
					&mut &justification[..],
				)
				.unwrap();
				total_justifications += 1;
				total_justifications_size += justification.encode().len();

				let votes_ancestries_len = justification.votes_ancestries.len();
				if votes_ancestries_len > max_justification_ancestry {
					max_justification_ancestry = votes_ancestries_len;
				}
			}

			if header.number() % 1024 == 0 {
				println!(
					"MaxHeader: total={} maximal={} average={:?}",
					total_maximal_headers,
					maximal_header_size,
					total_maximal_header_size.checked_div(total_maximal_headers)
				);
				println!(
					"AvgHeader: total={} maximal={} average={:?}",
					total_average_headers,
					maximal_average_header_size,
					total_average_header_size.checked_div(total_average_headers)
				);
				println!(
					"Justifica: total={} average={:?} max_votes={}",
					total_justifications,
					total_justifications_size.checked_div(total_justifications),
					max_justification_ancestry
				);
			}
		}
	});
```